e2eAIOK/DeNas/search/utils.py

```diff
@@ -22,4 +22,16 @@ def parse_config(conf_file):
         return settings
     with open(conf_file) as f:
         settings.update(yaml.safe_load(f))
-    return settings
\ No newline at end of file
+    return settings
+
+def timeout_input(printout, default, timeout = None, interactive = True):
+    if not interactive:
+        return default
+    import sys, select
+    print(printout)
+    i, o, e = select.select([sys.stdin], [], [], timeout)
+    if (i):
+        msg = sys.stdin.readline().strip()
+        return default if len(msg) == 0 else msg
+    else:
+        return default
```

Comment:
@zigzagcai , I saw this utils for common used func, can you put it to common lib?

Reply:
Sure. I will fix it.

Reply:
@xuechendi It has been fixed.